tests/lib/nested.sh

```diff
@@ -49,25 +49,54 @@ get_qemu_for_nested_vm(){
     command -v qemu-system-x86_64
 }
 
-get_image_url_for_nested_vm(){
+get_google_image_url_for_nested_vm(){
     case "$SPREAD_SYSTEM" in
-    ubuntu-16.04-64)
-        echo "https://cloud-images.ubuntu.com/xenial/current/xenial-server-cloudimg-amd64-disk1.img"
-        ;;
-    ubuntu-18.04-64)
-        echo "https://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img"
-        ;;
-    ubuntu-19.10-64)
-        echo "https://cloud-images.ubuntu.com/eoan/current/eoan-server-cloudimg-amd64.img"
-        ;;
-    ubuntu-20.04-64)
-        echo "https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img"
-        ;;
-    *)
-        echo "unsupported system"
-        exit 1
-        ;;
-    esac
+        ubuntu-16.04-64)
+            echo "https://storage.googleapis.com/spread-snapd-tests/images/xenial-server-cloudimg-amd64-disk1.img"
+            ;;
+        ubuntu-18.04-64)
+            echo "https://storage.googleapis.com/spread-snapd-tests/images/bionic-server-cloudimg-amd64.img"
+            ;;
+        ubuntu-19.10-64)
+            echo "https://storage.googleapis.com/spread-snapd-tests/images/eoan-server-cloudimg-amd64.img"
+            ;;
+        ubuntu-20.04-64)
+            echo "https://storage.googleapis.com/spread-snapd-tests/images/focal-server-cloudimg-amd64.img"
+            ;;
+        *)
+            echo "unsupported system"
+            exit 1
+            ;;
+        esac
+}
+
+get_ubuntu_image_url_for_nested_vm(){
+    case "$SPREAD_SYSTEM" in
+        ubuntu-16.04-64)
+            echo "https://cloud-images.ubuntu.com/xenial/current/xenial-server-cloudimg-amd64-disk1.img"
+            ;;
+        ubuntu-18.04-64)
+            echo "https://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img"
+            ;;
+        ubuntu-19.10-64)
+            echo "https://cloud-images.ubuntu.com/eoan/current/eoan-server-cloudimg-amd64.img"
+            ;;
+        ubuntu-20.04-64)
+            echo "https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img"
+            ;;
+        *)
+            echo "unsupported system"
+            exit 1
+            ;;
+        esac
+}
+
+get_image_url_for_nested_vm(){
+    if [ "$SPREAD_BACKEND" = "google" ]; then
```

Comment:
Shouldn't this work for "google" and "google-nested"?

Reply:
thanks for the review, in fact not just google-nested but also google-sru and google-tpm